src/xll_utils/mod.rs

```diff
@@ -0,0 +1,128 @@
+mod registration;
```

Comment:
This is a solid piece of work, should it be a separate crate? It seems general enough that others may wish to use it.